src/cli/commands/install.js

```diff
@@ -359,6 +359,13 @@ export class Install {
       return true;
     }
 
+    if (!match.integrityMatches && match.integrityError == 'PATTERNS_DONT_MATCH') {
+      this.reporter.warn(this.reporter.lang('integrityPatternsDontMatch'));
+      this.reporter.info(this.reporter.lang('uninstallRegenerate'));
+      this.flags.force = true;
```

Comment:
I think it should cleanup lockfile, yes.
Just force may mean more things like forcing rebuilding binary dependencies.